optima/results.py

```diff
@@ -1028,50 +1078,86 @@ def exporttoexcel(filename=None, outdict=None):
     workbook = Workbook(filename)
     
     for key,outstr in outdict.items():
-        worksheet = workbook.add_worksheet(sanitizefilename(key)) # A valid filename should also be a valid Excel key
+        if key == 'Optimal':
+            key = 'Optimized'  # Hack to replace optimal with optimized for sheet name
+        worksheet = workbook.add_worksheet(sanitizefilename(key))  # A valid filename should also be a valid Excel key
         
         # Define formatting
         budcovformats = ['Budget', 'Coverage', 'Time-varying']
-        colors = {'gentlegreen':'#3c7d3e', 'fadedstrawberry':'#ffeecb', 'edgyblue':'#bcd5ff','accountantgrey':'#f6f6f6', 'white':'#ffffff'}
+        colors = {'gentlegreen': '#3c7d3e', 'fadedstrawberry': '#ffeecb', 'edgyblue': '#bcd5ff',
+                  'accountantgrey': '#f6f6f6', 'white': '#ffffff',
+                  'increasegreen': '#c4d79b', 'decreasered': '#ffc7ce'}
         formats = dict()
         formats['plain'] = workbook.add_format({})
         formats['bold'] = workbook.add_format({'bg_color': colors['edgyblue'], 'bold': True})
-        formats['number'] = workbook.add_format({'bg_color': colors['fadedstrawberry'], 'num_format':0x04})
+        formats['number'] = workbook.add_format({'bg_color': colors['fadedstrawberry'], 'num_format': 0x04})
+        formats['percent'] = workbook.add_format({'bg_color': colors['fadedstrawberry'], 'num_format': 0x09})
         formats['header'] = workbook.add_format({'bg_color': colors['gentlegreen'], 'color': colors['white'], 'bold': True})
+        formats['increase'] = workbook.add_format({'bg_color': colors['increasegreen'], 'num_format': 0x09})
+        formats['decrease'] = workbook.add_format({'bg_color': colors['decreasered'], 'num_format': 0x09})
         
         # Convert from a string to a 2D array
         outlist = []
         for line in outstr.split('\n'):
             outlist.append([])
-            for cell in line.split(','):
-                if cell=='tot': cell = 'Total' # Hack to replace internal key with something more user-friendly
-                outlist[-1].append(str(cell)) # If unicode, doesn't work
+            if '"' in line:
+                for cell in line.split('","'):  # Hack to not separate program names with commas in them
+                    outlist[-1].append(str(cell))  # If unicode, doesn't work
+            else:
+                for cell in line.split(','):
+                    if cell == 'tot': cell = 'Total'  # Hack to replace internal key with something more user-friendly
+                    outlist[-1].append(str(cell))  # If unicode, doesn't work
```

Comment:
I'm not sure this logic is going to be very robust...the `if '"' in line` seems too broad, e.g. won't this break if you have a quote in a program name? I think a safer approach would be to do sanitization on the program names before they get here and strip out any commas (and possibly any other invalid characters).

Reply:
I added all of "," as a separator of cells only for program names (instead of just ,), but I agree it's hacky, not very robust, and could fail (if a program name contained "," for some reason...). Is there already a sanitizing code for strings/program names? If so, please feel free to change and commit on top of this.